base/applications/taskmgr/procpage.c

```diff
@@ -1150,69 +1149,66 @@ void ProcessPage_OnProperties(void)
     ShellExecuteExW(&info);
 
 Cleanup:
-
-    if (pszPath)
-    {
-        HeapFree(GetProcessHeap(), 0, pszPath);
-    }
+    HeapFree(GetProcessHeap(), 0, pszExePath);
 }
 
 void ProcessPage_OnOpenFileLocation(void)
 {
     DWORD dwProcessId;
-    WCHAR szPath[MAX_PATH];
-    LPWSTR pszPath = NULL;
-    LPWSTR pszExePath = NULL;
+    LPWSTR pszExePath;
     LPWSTR pszCmdLine = NULL;
     DWORD dwLength;
 
     dwProcessId = GetSelectedProcessId();
-    dwLength = GetProcessExecutablePathById(dwProcessId, szPath, _countof(szPath));
 
-    if (dwLength == 0)
+    /* Allocate a buffer for the path */
+    pszExePath = HeapAlloc(GetProcessHeap(), 0, MAX_PATH * sizeof(WCHAR));
```

Comment:
Since in all cases (this call path, or the one below if `GetProcessExecutablePathById()` returns a longer length than MAX_PATH) you dynamically allocate a buffer, why not just more simply do one single call to `GetProcessExecutablePathById()` in order to retrieve that length ?